src/Board.cpp

```diff
@@ -0,0 +1,185 @@
+#include "..\include\Board.h"
+
+Board::Board(const std::string& bStr): m_currentPlayerIsWhite(true)
+{
+    clearAndBuildBoard(bStr);
+}
+
+int Board::codeRespone(const std::string& move)
+{
+    
+
+    Position src = { move[0] - 'a', move[1] - '1' };
+    Position dst = { move[2] - 'a', move[3] - '1' };
+
+    if (isEmpty(src))
+        return 11;
+
+    if (isRival(src))
+        return 12;
+
+    if (isCurrentTurnPiece(dst))
+        return 13;
+    if (!m_board[src.x][src.y]->isAcceptableMove(src, dst) || isWayBloacked(src, dst))
+        return 21;
+
+    if (isPossibleChess(src, dst))
+        return 31;
+
+    update(src, dst);
+
+    if (isGameOver())
+        return 41;
+
+    return 42;
+
+}
+
+void Board::clearAndBuildBoard(const std::string& bStr)
+{
+    for (int row = 0; row < ROWS; ++row)
+        for (int col = 0; col < COLS; ++col)
+            m_board[row][col] = nullptr;

```

Comment:
Instead of manually setting each element of m_board to nullptr using a nested for loop, you can use std::fill or std::fill_n to initialize the entire m_board with nullptr in a more concise way. like this: std::fill_n(&m_board[0][0], ROWS * COLS, nullptr);